src/main/java/org/trigon/hardware/misc/leds/LEDStrip.java

```diff
@@ -0,0 +1,103 @@
+package org.trigon.hardware.misc.leds;
+
+import com.ctre.phoenix.led.LarsonAnimation;
+import edu.wpi.first.wpilibj.util.Color;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+import java.util.function.Supplier;
+
+public abstract class LEDStrip extends SubsystemBase {
+    public static LEDStrip[] LED_STRIPS = new LEDStrip[0];
+    final int indexOffset;
+    final boolean inverted;
+    final int numberOfLEDs;
+
+    public LEDStrip(boolean inverted, int numberOfLEDs, int indexOffset) {
+        this.inverted = inverted;
+        this.numberOfLEDs = numberOfLEDs;
+        this.indexOffset = indexOffset;
+
+        addLEDStripToLEDStripsArray(this);
+    }
+
+    public static void setDefaultCommandForAllLEDS(Command command) {
+        for (LEDStrip ledStrip : LED_STRIPS)
+            ledStrip.setDefaultCommand(command);
+    }
+
+    public int getNumberOfLEDS() {
+        return numberOfLEDs;
+    }
+
+    abstract void resetLEDSettings();
+
+    /**
+     * Sets the color of the LED strip to the given color.
+     *
+     * @param color the color to set the LED strip to
+     */
+    abstract void staticColor(Color color);
+
+    /**
+     * Blinks the LED strip between two colors.
+     *
+     * @param firstColor              the first color to blink
+     * @param secondColor             the second color to blink
+     * @param blinkingIntervalSeconds the interval in seconds to blink between the two colors
+     */
+    abstract void blink(Color firstColor, Color secondColor, double blinkingIntervalSeconds);
+
+    /**
+     * "Breathes" a bunch of LEDs with a given color.
+     *
+     * @param color            the color of the breathing LEDs
+     * @param breathingLEDs    the amount of breathing LEDs
+     * @param cycleTimeSeconds the time it takes for a full cycle of the breathing
+     * @param inverted         whether the breathing should be inverted
+     * @param bounceMode       the bounce mode of the breathing
```

Comment:
This is obvious. It needs to explain the actual use, not the variable name in different works